cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,216 @@
+                                "                \"StorageMetadataCheckpointOperation|StreamSegmentAppendOperation|StreamSegmentMapOperation|\" +\n" +
+                                "                \"StreamSegmentSealOperation|StreamSegmentTruncateOperation|UpdateAttributesOperation]");
+                        predicates.add( a -> a.contains(op));
+                        break;
+                    case "SequenceNumber":
+                        long in = getLongUserInput("Valid Sequence Number: ");
+                        predicates.add( a -> a.contains(Long.toString(in)));
+                        break;
+                    case "SegmentId":
+                        in = getLongUserInput("Valid segmentId to search: ");
+                        predicates.add( a -> a.contains(Long.toString(in)));
+                        break;
+                    case "Offset":
+                        in = getLongUserInput("IValid offset to seach: ");
+                        predicates.add( a -> a.contains(Long.toString(in)));
+                        break;
+                    case "Length":
+                        in = getLongUserInput("IValid length to seach: ");
+                        predicates.add( a -> a.contains(Long.toString(in)));
+                        break;
+                    case "Attributes":
+                        in = getLongUserInput("Valid number of attributes to seach: ");
+                        predicates.add( a -> a.contains(Long.toString(in)));
+                        break;
+                    default:
+                        output("Invalid operation, please select one of [delete|add|replace]");
+                }
+                predicate = clause.equals("and") ?
+                        predicates.stream().reduce(Predicate::and).orElse( x -> true) : predicates.stream().reduce(Predicate::or).orElse( x -> false);
+            } catch (NumberFormatException ex) {
+                outputError("Wrong input argument.");
+                outputException(ex);
+            } catch (IllegalStateException ex) {
+                // Last input was incorrect, so remove it.
+                output("Last Log Inspect Operation did not pass the checks, removing it from list.");
+                //durableLogEdits.remove(durableLogEdits.size() - 1);
+            } catch (Exception ex) {
+                outputError("Some problem has happened.");
+                outputException(ex);
+            }
+            output("You can continue adding conditions for inspect.");
+            finishInputCommands = !confirmContinue();
+            next = !finishInputCommands;
+        }
+        output("Value of predicates is : " + predicates);
+        return predicate;
+    }
+
+    public static CommandDescriptor descriptor() {
+        return new CommandDescriptor(COMPONENT, "durableLog-inspect", "Allows to repair DurableLog " +
```

Comment:
Change the description, it is a copy from the log repair command.

Reply:
done.